db/migrate/20230706033605_add_user_id_to_search_results.rb

```diff
@@ -0,0 +1,12 @@
+class AddUserIdToSearchResults < ActiveRecord::Migration[7.0]
+  def up
+    # Add user_id columns
+    add_column :search_results,    :user_id, :uuid, null: true
+
+    add_foreign_key :search_results, :users, column: :user_id
```

Comment:
Can we use `add_reference` to replace these lines?
```suggestion
    add_reference   :search_results, :users, foreign_key: true
```

Reply:
@hoangmirs 👍 I will need to add `type: :uuid` since the [default type is int ](https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference)so it doesn't match with user_id type.

```
add_reference   :search_results, :users, type: :uuid, foreign_key: true
```

Reply:
@hoangmirs I noticed the default column will be named `users_id` instead of `user_id`, it will affect the reference in the codebase and test classes, so I think it'd be better if we revert the update 🙏

Reply:
@hoangmirs 🙈 The reference_name should be `:user` instead of `:users`. Pushed 🚀